lib/sync/default-dataHandlers.js

```diff
@@ -43,7 +53,7 @@ module.exports = function (db) {
                      'doRead : ' + dataset_id +
                       ' :: ' + uid +
                       ' :: meta=' + util.inspect(meta_data));
-      mongo.collection(dataset_id).findOne({"_id" : ObjectID(uid)}, function(err, found){
+      mongo.collection(dataset_id).findOne({"_id" : convertToObjectId(dataset_id, uid)}, function(err, found){
```

Comment:
Just a question really. I am wondering whether we are handling when `found` here is `null` or `undefined` in sync? 

In the case where a non-ObjectID is returned by `convertToObjectId` both `err` and `found` will be `undefined` for mongodb.findOne, I think.

Do we handle this somewhere or will we expect the handler implementation to handle it? Or, does it even need to be handled?